CHANGELOG.md

```diff
@@ -6,6 +6,9 @@ The MSRV (Minimum Supported Rust Version) is 1.37.0, and typenum is tested
 against this Rust version.
 
 ### Unreleased
+- [removed] Remove `force_unix_path_separator` feature, make it the default
```

Comment:
Why is this removed?

IIRC, there was a reason for having it as optional. I think folks had issues both with it and without it, in different environments. I think it was related to cross-compilation involving Windows and/or Bazel?

Reply:
The original commit (694130d80cee6316879b1eb4889a7db1de3755b3) suggested that _enabling_ `force_unix_path_separator` improved the compatibility with non-Cargo build tools. It looks like the previous version only used the environment variables emitted by cargo to do the including. Effectively, `force_unix_path_separator` is now the default, with no environment variables, which is how every crate I've ever seen does things.

Furthermore, it interfered with the `doc_auto_cfg` feature. Instead of picking up the `const-generics` crate feature, it would use `force_unix_path_separator`, which was misleading and ugly.

Reply:
Did some digging.

Looks like Bazel can't process the custom environment variables: https://github.com/paholg/typenum/issues/127, so this improves the situation there.

However, it seems the solution without env variables breaks wasm-pack on windows: https://github.com/paholg/typenum/issues/124

Maybe wasm-pack works better now? I don't have a Windows dev machine, but I'll do some more digging and try to figure out if this is safe to remove.

Reply:
I've most certainly used wasm-pack in conjunction with this form of including in other crates. Will try to reproduce with typenum today. Includes are purely a rustc thing, too, not sure how it would affect wasm-pack.

However, don't let this hold up publishing a new version with the 32-bit build fixes. I've already published `generic-array` 1.0, with links in the readme/crate that link back to our Github Pages docs that includes typenum with `const-generics`, since the docs.rs build was broken.

Reply:
I'm on a Windows host, using `wasm-pack new test-project`

Adding to `Cargo.toml`:
```toml
[patch.crates-io]
typenum = { git = "https://github.com/novacrazy/typenum" }
```

then adding some code that uses `generic-array 0.14`, running `wasm-pack build` builds/runs totally fine. If I open up the same project on a Linux VM, it also builds totally fine. 

I'm unsure if there ever was an actual issue?

Reply:
Hm, okay. Let's ship it!

Reply:
It's possible Rust itself updated at some point to fix paths automatically since 2018, perhaps https://github.com/rust-lang/rust/pull/89174/ in 1.58.0 in 2022, or some other minor fix somewhere.